pylint/checkers/base.py

```diff
@@ -838,8 +836,7 @@ def _check_else_on_loop(self, node):
 
     def _check_in_loop(self, node, node_name):
         """check that a node is inside a for or while loop"""
-        _node = node.parent
-        while _node:
+        for _node in node.node_ancestors():
```

Comment:
```suggestion
        for parent in node.node_ancestors():
```
I would suggest to be consistent with it and use `parent` whenever possible.

Reply:
My first thought was to make the changes while disturbing the rest of the code as little as possible. But the variables are renamed now.